pkg/debug/checkups/services_windows.go

```diff
@@ -0,0 +1,285 @@
+		s.data["config"] = fmt.Sprintf("error: %v", err)
+	} else {
+		s.data["binary_path_name"] = cfg.BinaryPathName
+		s.data["load_order_group"] = cfg.LoadOrderGroup
+		s.data["dependencies"] = cfg.Dependencies
+		s.data["service_start_name"] = cfg.ServiceStartName
+		s.data["display_name"] = cfg.DisplayName
+		s.data["description"] = cfg.Description
+		s.data["delayed_auto_start"] = cfg.DelayedAutoStart
+		s.data["tag_id"] = cfg.TagId
+
+		s.data["service_type_raw"] = cfg.ServiceType
+		s.data["service_type"] = serviceTypeHumanReadable(cfg.ServiceType)
+
+		s.data["start_type_raw"] = cfg.StartType
+		s.data["start_type"] = startTypeHumanReadable(cfg.StartType)
+
+		s.data["error_control_raw"] = cfg.ErrorControl
+		s.data["error_control"] = errorControlHumanReadable(cfg.ErrorControl)
+
+		s.data["sid_type_raw"] = cfg.SidType
+		s.data["sid_type"] = sidTypeHumanReadable(cfg.SidType)
+	}
+
+	// If we're running doctor and not flare, no need to write extra data
+	if extraWriter == nil {
+		return nil
+	}
+
+	extraZip := zip.NewWriter(extraWriter)
+	defer extraZip.Close()
+
+	if err := gatherServices(extraZip, serviceManager); err != nil {
+		return fmt.Errorf("gathering service list: %w", err)
+	}
+
+	if err := gatherServiceManagerEventLogs(ctx, extraZip); err != nil {
+		return fmt.Errorf("gathering service manager event logs: %w", err)
+	}
+
+	return nil
+}
+
+func stateHumanReadable(state svc.State) string {
+	// For mapping, see https://pkg.go.dev/golang.org/x/sys/windows/svc#pkg-constants
+	switch state {
+	case svc.Running:
+		return "SERVICE_RUNNING"
+	case svc.Stopped:
+		return "SERVICE_STOPPED"
+	case svc.StartPending:
+		return "SERVICE_START_PENDING"
+	case svc.StopPending:
+		return "SERVICE_STOP_PENDING"
+	case svc.ContinuePending:
+		return "SERVICE_CONTINUE_PENDING"
+	case svc.PausePending:
+		return "SERVICE_PAUSE_PENDING"
+	case svc.Paused:
+		return "SERVICE_PAUSED"
+	default:
+		return fmt.Sprintf("unknown: %d", state)
+	}
+}
+
+func serviceTypeHumanReadable(serviceType uint32) string {
+	// For mapping, see https://pkg.go.dev/golang.org/x/sys@v0.11.0/windows#SERVICE_KERNEL_DRIVER
+	switch serviceType {
+	case 1:
+		return "SERVICE_KERNEL_DRIVER"
+	case 2:
+		return "SERVICE_FILE_SYSTEM_DRIVER"
+	case 4:
+		return "SERVICE_ADAPTER"
+	case 8:
+		return "SERVICE_RECOGNIZER_DRIVER"
+	case 16:
+		return "SERVICE_WIN32_OWN_PROCESS"
+	case 32:
+		return "SERVICE_WIN32_SHARE_PROCESS"
+	case 16 | 32:
+		return "SERVICE_WIN32"
+	case 256:
+		return "SERVICE_INTERACTIVE_PROCESS"
+	case 1 | 2 | 8:
+		return "SERVICE_DRIVER"
+	case 1 | 2 | 4 | 8 | 16 | 32 | 256:
+		return "SERVICE_TYPE_ALL"
+	default:
+		return fmt.Sprintf("unknown: %d", serviceType)
+	}
+}
+
+func startTypeHumanReadable(startType uint32) string {
+	// For mapping, see https://pkg.go.dev/golang.org/x/sys/windows/svc/mgr#pkg-constants
+	// and https://pkg.go.dev/golang.org/x/sys@v0.11.0/windows#SERVICE_BOOT_START
+	switch startType {
+	case 0:
```

Comment:
These are constants in the windows package, should we use them? `case windows.SERVICE_BOOT_START` or something?

Reply:
Oh yup that would improve readability a lot -- thank you, I'll update